src/Frontend/Core.hs

```diff
@@ -247,13 +251,15 @@ redirectFormHandler getPage processor = getH :<|> postH
         v <- getForm fa (processor1 page)
         renderer page v fa
 
-    postH env = do
+    postH formData = do
         page <- getPage
         let fa = absoluteUriPath $ formAction page
+            env = getFormDataEnv formData
         (v, mpayload) <- postForm fa (processor1 page) (\_ -> return $ return . runIdentity . env)
-        case mpayload of
+        (case mpayload of
             Just payload -> processor2 page payload >>= redirect
-            Nothing      -> renderer page v fa
+            Nothing      -> renderer page v fa)
+            `finally` cleanupTempCsvFiles formData
```

Comment:
Note that as long as we read (or write to) files eagerly and close them in obviously safe places (e.g., a parent thread of all potentially file-opening threads, after they all terminate), we don't need to use `resourceForkIO`, which is one of the main complexities of the `resourcet` engine and it's use pattern. Let's keep it that way.